src/main/java/uk/gov/hmcts/darts/retention/enums/RetentionConfidenceScoreEnum.java

```diff
@@ -7,7 +7,7 @@
 @RequiredArgsConstructor
 public enum RetentionConfidenceScoreEnum {
 
-    CASE_PERFECTLY_CLOSED(1), CASE_NOT_PERFECTLY_CLOSED(2);
+    CASE_PERFECTLY_CLOSED(0), CASE_NOT_PERFECTLY_CLOSED(1);
```

Comment:
Should this PR include a flyway script to migrate `1` -> `0`, and `2` -> `1` for `court_case.ret_conf_score`, otherwise we'll have problems with the existing case data being wrongly interpreted?

Reply:
Were these missed previously? I can't see any entries in the database in the retention_confidence_category_mapper table. Don't think what you have done will work without those?

Reply:
`retention_confidence_category_mapper` is indeed empty, feedback from @hemantasharma1129 is that the mapping table will be populated as a separate task. If a mapping from a category doesn't exist (e.g. because the `retention_confidence_category_mapper` table is empty), then `null` values will get populated for score and reason against the case.

`court_case.ret_conf_score` however has numerical values already populated for various cases in the DB, and my understanding is we're changing the meaning of those numerical values in this PR. So any existing case records that have `ret_conf_score=1` would have intended that to mean `CASE_PERFECTLY_CLOSED`. However when this PR goes in, they will be interpreted as `CASE_NOT_PERFECTLY_CLOSED`, which seems dangerous?

Reply:
@chrisbellingham-hmcts yes we can introduce a flyway script to update 1 -> 0 and 2 -> 1 any existing data in DB.

@jackmaloney retention_confidence_category_mapper is part of Retention migration Path to Live item 36. This is agreed that RCC development will be completed from our end so that nothing is pending  and test RCC once the mapper table is populated by migration. As there is no data in the mapper table, no reasons will be populated. I will check if a  Test ticket exist, if not I will create one.

Reply:
Flyway script added in commit https://github.com/hmcts/darts-api/pull/2382/commits/7546b060012ca5ed516a776f2b61b2f8ea26e62d for all impacted tables including:

- `annotation_document`
- `case_document`
- `court_case`
- `media`
- `retention_confidence_category_mapper` (this table should be empty, but it's good form to include the migration anyway)
- `transcription_document`

Reply:
Are we saying that this ticket cannot be tested util migration do that @hemantasharma1129 ? Why do we need migration team to populate this table? We have other reference data tables that application team are the owners of that migration team won't migrate, can we not do this for this table?

Reply:
This ticket is tested with hardcoded value already. Those hardcoded values need to be persisted so that migration can also refer to it and be in sync. The reason we cant populate these values because we dont know what the ret_conf_reason values would be. Only key we have is the Confidence Category (1,2 & 3). Judith did an analysis which Jonathan will continue. This table will be populated part of Retention migration which is due in early Jan 2025.

Reply:
Confirmed this will need to be done in new year by migration so i'll close this one.